File: snippets/cpp/VS_Snippets_CLR/Generic.SortedList/cpp/remarks.cpp
#using <System.dll>

using namespace System;
using namespace System::Collections::Generic;

public ref class Example
{
public:
    static void Main()
    {
        // Create a new sorted list of strings, with string
        // keys.
        SortedList<int, String^>^ mySortedList =
            gcnew SortedList<int, String^>();

        // Add some elements to the list. There are no
        // duplicate keys, but some of the values are duplicates.
        mySortedList->Add(0, "notepad.exe");
        mySortedList->Add(1, "paint.exe");
        mySortedList->Add(2, "paint.exe");
        mySortedList->Add(3, "wordpad.exe");

        //<Snippet11>
        String^ v = mySortedList->Values[3];
        //</Snippet11>

        Console::WriteLine("Value at index 3: {0}", v);

        //<Snippet12>
        for each( KeyValuePair<int, String^> kvp in mySortedList )
        {
            Console::WriteLine("Key = {0}, Value = {1}", kvp.Key, kvp.Value);
        }
        //</Snippet12>
    }
};

int main()
{
    Example::Main();
}

File: snippets/csharp/System.Collections/Hashtable/Overview/remarks.cs
using System;
using System.Collections;

public class Remarks
{
    public static void Run()
    {
        // Create a new hash table.
        //
        Hashtable myHashtable = new()
        {
            // Add some elements to the hash table. There are no
            // duplicate keys, but some of the values are duplicates.
            { "txt", "notepad.exe" },
            { "bmp", "paint.exe" },
            { "dib", "paint.exe" },
            { "rtf", "wordpad.exe" }
        };

        // When you use foreach to enumerate hash table elements,
        // the elements are retrieved as KeyValuePair objects.
        // <snippet01>
        foreach (DictionaryEntry de in myHashtable)
        {
            // ...
        }
        // </snippet01>
    }
}
